tests/test_body_match.js

```diff
@@ -48,6 +48,105 @@ test('match body with regex inside array', function (t) {
   });
 })
 
```

Comment:
The tests you've add are all related and nearly identical, with small variation for the different payloads tested. Could you please put all the tests you've added within a test block for organization, eg:
```
test('body matching', function (t) {
  t.test('fails for invalid regex inside array', function (t2) {
     // Do work
     t.done();
  });
});
```

Another improvement would be to create a function to dynamically generate your tests, eg

```
var testBodyMatch = function (expected, actual, shouldMatch, t) {
  nock('http://encodingsareus.com')
    .post('/', expected)
    .reply(200);

  mikealRequest({
  url: 'http://encodingsareus.com/',
  method: 'post',
  json: actual,
}, function(err, res) {
  shouldMatch 
    ? assert.equal(res.statusCode, 200)
    : assert(err && /No match for request/.test(err.message), 'nock should not intersept the request');
    t.end();
});
```